pkg/modules/sentry/sentry.go

```diff
@@ -0,0 +1,162 @@
+			fs.Duration("sentry-flush-timeout", time.Duration(2)*time.Second, "Set the time limit (seconds) to wait for the underlying Sentry Transport to send any buffered events to Sentry server during Gotenberg stop")
+			return fs
+		}(),
+		New: func() gotenberg.Module { return new(Sentry) },
+	}
+}
+
+// Provision configures the Sentry module by reading flags.
+func (s *Sentry) Provision(ctx *gotenberg.Context) error {
+	flags := ctx.ParsedFlags()
+
+	// Logger.
+	loggerProvider, err := ctx.Module(new(gotenberg.LoggerProvider))
+	if err != nil {
+		return fmt.Errorf("get logger provider: %w", err)
+	}
+
+	logger, err := loggerProvider.(gotenberg.LoggerProvider).Logger(s)
+	if err != nil {
+		return fmt.Errorf("get logger: %w", err)
+	}
+
+	s.logger = logger
+
+	// Populate Sentry configuration from flags.
+	s.dsn = flags.MustString("sentry-dsn")
+	s.sendDefaultPii = flags.MustBool("sentry-send-default-pii")
+	s.environment = flags.MustString("sentry-environment")
+	s.flushTimeout = flags.MustDuration("sentry-flush-timeout")
+
+	// If no dsn is provided, Sentry integration is disabled.
+	if s.dsn == "" {
+		return nil
+	}
+
+	// Prepare Sentry client options.
+	s.sentryClientOptions = sentry.ClientOptions{
+		Dsn:            s.dsn,
+		SendDefaultPII: s.sendDefaultPii,
+	}
+
+	// Set environment if defined.
+	if s.environment != "" {
+		s.sentryClientOptions.Environment = s.environment
+	}
+
+	// Enable Sentry SDK debug logging if the application's log level is debug.
+	if s.logger.Core().Enabled(zapcore.DebugLevel) {
+		s.sentryClientOptions.Debug = true
+	}
+
+	// Used to track if Sentry client was initialized in Start() which allows Stop() to correctly flush any buffered events.
+	s.sentryInitialized = false
+
+	return nil
+}
+
+// Debug returns additional debug data.
+func (s *Sentry) Debug() map[string]interface{} {
+	debug := make(map[string]interface{})
+
+	debug["version"] = fmt.Sprintf("Sentry Go SDK Version: %s", sentry.SDKVersion)
+
+	return debug
+}
+
+// Start initializing the Sentry SDK.
+func (s *Sentry) Start() error {
+	if s.sentryClientOptions.Dsn != "" {
+		err := sentry.Init(s.sentryClientOptions)
+		if err != nil {
+			// This error is reported if a dsn was provided but Sentry failed to initialize.
+			return fmt.Errorf("Sentry configuration error: %w", err)
+		}
+
+		s.sentryInitialized = true
+	}
+
+	return nil
+}
+
+// StartupMessage returns a custom startup message.
+func (s *Sentry) StartupMessage() string {
+	if s.sentryInitialized {
+		return "initialized successfully"
+	}
+	return "integration is disabled - no DSN provided"
+}
+
+// Stop flushes any remaining events.
+func (s *Sentry) Stop(ctx context.Context) error {
+	// Flush buffered events before the program terminates, but only when Sentry client is initialized.
+	// Set the timeout to the maximum duration the program can afford to wait.
+	if s.sentryInitialized {
+		defer sentry.Flush(s.flushTimeout)
```

Comment:
```suggestion
		defer sentry.Flush(s.flushTimeout)
```

I think we should use the context deadline 🤔

Reply:
@gulien thanks for the comment - I don't know what that is or how to do it. Could you please advise or provide a bit more context?

Reply:
@gulien are you thinking of something like this where we can compute the remaining time on the deadline? If this was the case, I'd set the default of `flushTimeout` to `10s` and then a minimal positive timeout of `1s` if the context is done or the deadline passed.

Your feedback on this is appreciated.

```go
// Stop flushes any remaining events.
func (s *Sentry) Stop(ctx context.Context) error {
	// Flush buffered events before the program terminates, but only when Sentry client is initialized.
	if s.sentryInitialized {
		// Defer the flush operation. The timeout will be determined based on the context
		// when the deferred function executes.
		defer func(stopCtx context.Context) {
			// Start with the configured Sentry flush timeout as the default/upper bound.
			timeoutForFlush := s.flushTimeout

			// If the Stop context has a deadline, the flush must not exceed
			// the time remaining for that deadline.
			if deadline, ok := stopCtx.Deadline(); ok {
				remainingTimeForContext := time.Until(deadline)
				if remainingTimeForContext < timeoutForFlush {
					// Context offers less time than s.flushTimeout, so use the context's remaining time.
					timeoutForFlush = remainingTimeForContext
				}
			}

			// If the context is already marked as "done" (e.g., canceled, or deadline already passed),
			// or if the calculated timeout ended up being non-positive (deadline passed during Stop execution),
			// then use a very small positive timeout for a best-effort, non-blocking flush.
			if err := stopCtx.Err(); err != nil {
				s.logger.Debug("Sentry Stop: context is already done, using minimal flush timeout.", zap.Error(err))
				timeoutForFlush = 1 * time.Second // Minimal positive timeout
			} else if timeoutForFlush <= 0 {
				s.logger.Debug("Sentry Stop: calculated flush timeout is non-positive (deadline likely passed), using minimal flush timeout.")
				timeoutForFlush = 1 * time.Second // Minimal positive timeout
			}

			s.logger.Debug("Sentry: flushing events", zap.Duration("timeout", timeoutForFlush))
			sentry.Flush(timeoutForFlush)
		}(ctx) // Pass the context to the deferred function's closure
	}

	return nil
}
```

Reply:
@gulien code added for this suggestion:

https://github.com/gotenberg/gotenberg/pull/1237/files#diff-48f5b89967188c919bf21913391a6283dde27a59b9cf424dfaec728ca7cfd5f3R132